src/stan/lang/grammars/semantic_actions_def.cpp

```diff
@@ -2534,7 +2544,7 @@ namespace stan {
     boost::phoenix::function<set_int_range_upper> set_int_range_upper_f;
 
     void validate_int_data_expr::operator()(const expression& expr,
-                                            int var_origin, bool& pass,
+                                            var_origin var_origin, bool& pass,
```

Comment:
changed to const reference.  also changed all var_origin variables named "var_origin" to "vo".  other functions in the code have var_origin variables named "origin" - kept these unchanged.  in two functions, changed name "origin_allowed" to "origin" so that usage across this file is more consistent.